src/blocks/folder-blocks/infinite-canvas/index.tsx

```diff
@@ -0,0 +1,282 @@
+import "./index.css";
+
+const width = 5000;
+const height = 5000;
+const defaultDimensions = [200, 100];
+export default function (
+  props: FolderBlockProps & {
+    metadata: { items: ItemType[] };
+  }
+) {
+  const {
+    context,
+    tree,
+    metadata,
+    BlockComponent,
+    onUpdateMetadata,
+    onRequestGitHubData,
+  } = props;
+
+  const wrapperElement = useRef<HTMLDivElement>(null);
+
+  const [pan, setPan] = useState<Dimensions>([-width / 2, -height / 2]);
+  useEffect(() => {
+    // onWheel needs to be in useEffect to make non-passive
+    // to prevent scroll from changing pages (left/right)
+    if (!wrapperElement?.current) return;
+    const onWheel = (e: WheelEvent) => {
+      setPan((pan) => {
+        const newPan = [pan[0] - e.deltaX, pan[1] - e.deltaY] as Dimensions;
+        return newPan;
+      });
+      e.preventDefault();
+    };
+
+    wrapperElement?.current.addEventListener("wheel", onWheel, {
+      passive: false,
+    });
+    return () => {
+      if (!wrapperElement?.current) return;
+      wrapperElement?.current.removeEventListener("wheel", onWheel);
+    };
+  }, []);
+
+  const { isDragging } = useDrag(wrapperElement, pan, (newPan) => {
+    const boundedDimensions = [
+      Math.min(Math.max(newPan[0], -width), 0),
+      Math.min(Math.max(newPan[1], -height), 0),
+    ] as Dimensions;
+    setPan(boundedDimensions);
+  });
+
+  const [items, setItems] = useState<ItemType[]>([]);
+  const [isDirty, setIsDirty] = useState(false);
+
+  const [fileContents, setFileContents] = useState<Record<string, string>>({});
+  const files = useMemo(
+    () => getNestedFileTree(tree)[0].children.filter((d) => d.type === "blob"),
+    [tree]
+  );
+
+  const getFileContents = async (path: string) => {
+    const url = `/repos/${context.owner}/${context.repo}/contents/${path}`;
+    const data = await onRequestGitHubData(url);
+    if (!data.content) {
+      return;
+    }
+    const decodedContent = Buffer.from(data.content, "base64").toString("utf8");
+    return decodedContent;
+  };
+
+  useEffect(() => {
+    items.forEach(async (item) => {
+      const path = item.path;
+      if (!path) return;
+      if (fileContents[path]) return;
+      setFileContents((fileContents) => ({
+        ...fileContents,
+        [path]: "Loading...",
+      }));
+      const content = await getFileContents(path);
+      if (!content) return;
+      setFileContents((fileContents) => ({ ...fileContents, [path]: content }));
+    });
+  }, [items]);
+
+  useEffect(() => {
+    setItems(metadata.items || placeholderItems);
+  }, [metadata]);
+
+  const [blockOptions, setBlockOptions] = useState<any[]>([]);
+
+  const getBlocks = async () => {
+    const url = "https://blocks-marketplace.githubnext.com/api/blocks";
+    const res = await fetch(url).then((res) => res.json());
+    const exampleBlocks =
+      res.find((d: any) => d.full_name === "githubnext/blocks-examples")
+        ?.blocks || [];
```

Comment:
why allow only example blocks here?

Reply:
that's a great question! let them all in!